src/DataStructures/LeviCivitaIterator.cpp

```diff
@@ -0,0 +1,65 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "DataStructures/LeviCivitaIterator.hpp"
+
+#include "Utilities/Gsl.hpp"
+
+namespace {
+template <size_t Dim>
+constexpr void initialize_indexes_and_signs(
+    gsl::not_null<std::array<Index<Dim>, factorial(Dim)>*> indexes,
+    gsl::not_null<std::array<int, factorial(Dim)>*> signs) noexcept {
+  auto index = Index<Dim>();
+  std::iota(index.begin(), index.end(), 0);
+
+  // Add the initial index permutation to the list
+  // Also add the sign, which is always +1, because std::iota
+  // initializes the indexes as 0,1,...Dim-1
+  (*indexes)[0] = index;
+  (*signs)[0] = 1;
+
+  size_t permutation_count = 1;
+  auto factoradic_counter = std::array<size_t, Dim>();
+
+  // std::next_permutation generates the different permuations of index,
+  // but how do you know whether the corresponding Levi Civita symbol is
+  // +1 or -1? To find out, compute the number, in the factorial number system,
```

Comment:
"system, also called factoradic..." to give context to your choice of variable name